Rename the override var for secret name

ttung opened the pull request:
### Description
It's not just for auth0.

### Test plan
git grep AUTH0_CONFIG_SECRET_NAME

jgadling reviewed (approved):
Headsup that this will break batch jobs for a little bit until I get a new base image built. Should only be a few minutes.

Commit:
Rename the override var for secret name

### Description
It's not just for auth0.

### Test plan
git grep AUTH0_CONFIG_SECRET_NAME

jgadling committed:
Make sure batch supports env var change.

jgadling commented:
Added a patch to this branch that means this shouldn't break batch anymore
